### ISSUE TITLE
Controller collection should not be bind with the view by name

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-ui#91]
As far as I understand controller collections bind with view regions by the same name.

I think it is wrong, because of two reasons.

First: it looks like magic, I was looking for some code which binds this collection for a while before I understand that they are bind behind the scene, by names.

Second: I understand controllers as code which glue two separate APIs: model and view. Expecting that there will be a collection with the same names looks wrong.

**Comment by Reinmar** [comment]
There's one more thing wrong here – collections of controller collections. This is super confusing.

We should try to achieve something like:

```js
class SomeController extends Controller {
    constructor() {
       this.items = new ControllerCollection( this.view.items ); // Maps ctrl items into view items.

       // or, if the parent controller must know about its children:
       this.addCollection( 'items' );
```

What's more, since the `SomeController#items` would often be mapped into some other collection (model), this could perhaps even look like this:


```js
// or this.bind()
this.map( this.items ).to( this.view.items, ( itemModel ) => {
    return new Controller( itemModel, new ItemView() );
} );
```

Of course, these are only some wild ideas. We need to understand whether they can be translated into something useful and something that can work.

**Comment by pjasiun** [comment]
Yes, I can confirm t that collection of the controllers collections in very confusing and calling:
```
collection.add( 'items', presence );
```
Instead:
```
collection.items.add( presence );
```
Is very confusing.

**Comment by Reinmar** [comment]
> collection.items.add( presence );

I think you meant:

```js
controller.items.add( presence );
```

**Comment by Reinmar** [comment]
Fixed by #100.
